### ISSUE TITLE
UDX stuck on start

### ISSUE THREAD

**Issue by jose1711** [UltraStar-Deluxe/USDX#395]
### Actual behaviour

```
$ ultrastardx
ALSA lib pcm_dsnoop.c:638:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dmix.c:1099:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm.c:2565:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2565:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2565:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
Expression 'alsa_snd_pcm_hw_params_set_period_size_near( pcm, hwParams, &alsaPeriodFrames, &dir )' failed in 'src/hostapi/alsa/pa_linux_alsa.c', line: 923
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_dmix.c:1099:(snd_pcm_dmix_open) unable to open slave
Expression 'alsa_snd_pcm_hw_params_set_period_size_near( pcm, hwParams, &alsaPeriodFrames, &dir )' failed in 'src/hostapi/alsa/pa_linux_alsa.c', line: 923
cannot connect alsa_pcm:capture_1 to alsa-jack.jackC.15645.3:in_000
Expression 'alsa_snd_pcm_start( stream->capture.pcm )' failed in 'src/hostapi/alsa/pa_linux_alsa.c', line: 2929
Expression 'AlsaStart( stream, 0 )' failed in 'src/hostapi/alsa/pa_linux_alsa.c', line: 4228
```

### Expected behaviour

UDX window should appear

### Steps to reproduce

Jackd server is started, test using `mplayer -ao jack` is successful (song is played).

### Details

Provide some additional information:

- USDX version: r831.387fd92a-1
- Operating System + version: Arch Linux 64bit

![obrazok](https://user-images.githubusercontent.com/1406222/47998604-447b9700-e100-11e8-852b-d8f28d1001c6.png)
```
~ ps aux|grep '[j]ack'
jose      2942  0.2  2.4 775212 149136 pts/0   SLl+ 12:12   0:14 qjackctl
jose      2954  4.9  1.5 253808 95056 ?        SLsl 12:12   4:22 /usr/bin/jackdbus auto
```

**Comment by jose1711** [comment]
UDX is now more waiting indefinitely on the start when built with `--enable-debug` and goes right into menus. no sound via jack but since it's now working normally using alsa i can get by. closing.

**Comment by s09bQ5** [comment]
There is a line `{$IF Defined(Debug) and Defined(Linux)}` in src/media/UAudioCore_Portaudio.pas.
Can you change that line so that it uses the else branch?

**Comment by jose1711** [comment]
with `--enable-debug` and only left `else` branch instead of condition checking UDX no longer hangs on startup but issues an error message and terminates:
```
LANG=C ultrastardx
ERROR:  no texture type for LoadingStaticAnimation found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic1Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic2Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic3Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic4Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic5Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic6Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic7 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic7Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic8 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic8Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic9 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic9Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic10 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic10Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic11 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic11Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic12 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for NamePlayerSelectStatic12Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SongRouletteStaticCat found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for JukeboxTimeProgress found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for JukeboxSongBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for JukeboxSongCover found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for JukeboxStaticActualSongCover found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for JukeboxSongMenuTimeProgress found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingTimeProgress found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1SingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1TwoPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1ThreePSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2RSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2MSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3SingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1Avatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1TwoPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1ThreePAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2RAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2MAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3RAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP1ThreePStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP1ThreePAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP2MStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP2MAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP3RStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP3RAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP1ThreePSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP2MSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingDuetP3RSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1FourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1FourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2FourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2FourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3FourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3FourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4FourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4FourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1SixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1SixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2SixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2SixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3SixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3SixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4SixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4SixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP5SixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP5SixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP6SixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP6SixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1DuetFourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1DuetFourPStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1DuetFourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2DuetFourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2DuetFourPStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2DuetFourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3DuetFourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3DuetFourPStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3DuetFourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4DuetFourPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4DuetFourPStatic2 found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4DuetFourPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1DuetSixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP1DuetSixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2DuetSixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP2DuetSixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3DuetSixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP3DuetSixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4DuetSixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP4DuetSixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP5DuetSixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP5DuetSixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP6DuetSixPSingBar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for SingP6DuetSixPAvatar found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubHeaderBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubCurrentNoteInfoBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubVolumeSliderBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubNotesBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubP1InfoBarBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubP2InfoBarBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for EditSubSentenceBackground found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for ScoreDownloadPopupProgressSong found. [TTheme.ThemeLoadStatic]
ERROR:  no texture type for ScoreDownloadPopupProgressWeb found. [TTheme.ThemeLoadStatic]
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pulse.c:243:(pulse_connect) PulseAudio: Unable to connect: Connection refused

ALSA lib pulse.c:243:(pulse_connect) PulseAudio: Unable to connect: Connection refused

ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
An unhandled exception occurred at $00007F25250A23C3:
EAccessViolation: Access violation
  $00007F25250A23C3
```
full error log is attached.
[Error.log](https://github.com/UltraStar-Deluxe/USDX/files/2560720/Error.log)

**Comment by tcrass** [comment]
Similar problem here (commit 9bd876e):

```

ALSA lib pcm_dsnoop.c:638:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dmix.c:1108:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm.c:2565:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2565:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2565:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_a52.c:823:(_snd_pcm_a52_open) a52 is only for playback
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_dmix.c:1108:(snd_pcm_dmix_open) unable to open slave
cannot connect alsa_pcm:capture_1 to alsa-jack.jackC.12405.3:in_000
Expression 'alsa_snd_pcm_start( stream->capture.pcm )' failed in 'src/hostapi/alsa/pa_linux_alsa.c', line: 2934
Expression 'AlsaStart( stream, 0 )' failed in 'src/hostapi/alsa/pa_linux_alsa.c', line: 4233
```

A did a "make clean", changed src/media/UAudioCore_Portaudio.pas like in

```
diff --git a/src/media/UAudioCore_Portaudio.pas b/src/media/UAudioCore_Portaudio.pas
index 2c142fc..25b30d8 100644
--- a/src/media/UAudioCore_Portaudio.pas
+++ b/src/media/UAudioCore_Portaudio.pas
@@ -300,8 +300,6 @@ begin
   // gdb's stops with the message: "ptrace: no process found".
   // Probably because the callback-thread is killed what confuses gdb.
   {$IF Defined(Debug) and Defined(Linux)}
-  cbWorks := true;
-  {$ELSE}
   // open device for testing
   err := Pa_OpenStream(stream, inParams, outParams, sampleRate,
           paFramesPerBufferUnspecified,

```

and recompiled, but didn't observer any difference -- game/ultrastardx still hangs with exactly the same messages as above, no matter whether it's called with or without --enable-debug.

Any news on this issue?

Cheers --

  Torsten

**Comment by barbeque-squared** [comment]
It is likely that the underlying issue has been addressed with the merging of #1231
